### ISSUE TITLE
class.BareName at beginning of line with one tab adds extra whitespace

### ISSUE THREAD

**Issue by washamdev** [NeVeSpl/NTypewriter#73]
Can anyone help me address this problem? The template contains code at one tab indentation. I've got a line that begins with `{{ class.BareName }}` on a new line:

> ![image](https://user-images.githubusercontent.com/4692325/230427957-f1c7b6ec-9460-4159-8ded-5567e1f7e5f2.png)

But when that template is rendered, it contains an extra tab:

> ![image](https://user-images.githubusercontent.com/4692325/230428214-f79a137b-62db-48cc-8a5c-13d1b78ec404.png)

How do I get that line to start with the right amount of indentation?

Just to be clear, we use four spaces as tabs.

**Comment by washamdev** [comment]
That removes all whitespaces _and_ new lines before that line though.
> ![image](https://user-images.githubusercontent.com/4692325/230438439-50e52b55-e2eb-42e0-bfd9-b6455dadeff5.png)

**Comment by NeVeSpl** [comment]
I have another idea, maybe the class name contains whitespace?  I would try
`{{ class.BareName | String.Strip }}`

**Comment by gregveres** [comment]
or maybe add a ``` {{ }}``` ahead of the ``` {{ - class.BareName }}``` to ensure that the new line isn't stripped.
In my templates, I make sure there are no indentations in the main body of my template. If I need to output something from a function, I return it as a string that the main body prints out. This way I control all the indentation from the main body. 

Like this:

```
{{- # Helper classes }}
{{- func ImportType(type)
	useType = Type.Unwrap(type)
	if (useType.ArrayType != null) 
	  useType = useType.ArrayType
	end
	if (type.IsEnum) 
	  path = "@skycourt/api/enums"
	else
	  path = "./" | String.Append useType.Name
    end
	if ((useType.Attributes | Array.Filter @AttrIsExportToTypescript).size > 0)
		ret "import { " | String.Append useType.Name | String.Append " } from '" | String.Append path | String.Append "';\r"
	end
	ret null
	end
}}
{{- func AttrIsExportToTypescript(attr)
	ret attr.Name | String.Contains "ExportToTypescript"
	end
}}

{{- # output classes }}
{{ $barrelFile = "" }}
{{- for class in data.Classes | Symbols.ThatHaveAttribute "ExportToTypescript" | Array.Concat (data.Classes | Symbols.ThatHaveAttribute "ExportToTypescriptWithKnockout") | Array.Sort "Name"
		capture output }}

{{- for type in (class | Type.AllReferencedTypes)}}
	{{- ImportType type}}
{{-end}}

export interface {{class.Name}}{{if class.HasBaseClass}} extends {{class.BaseClass.Name; end}} {
  {{- for prop in class.Properties | Symbols.ThatArePublic }}
  {{ prop.Name | String.ToCamelCase }}: {{prop | Custom.ToTypeScriptType }}{{if !for.last}},{{end}}
  {{-end}}
}
{{- end}}
```

**Comment by washamdev** [comment]
@gregveres The `{{ }}{{- class.BareName }}` worked! Thank you! Is this a bug or expected behavior?

**Comment by NeVeSpl** [comment]
Obviously, it is not expected, but Scriban is responsible for that part. NTypewriter uses Scriban v5.5, and apparently, there was some fix in version 5.6 of scriban/scriban#483 related to not intended indentation.

**Comment by gregveres** [comment]
Based on the documentation, I would say it is working as expected. 
But if it is a bug, then it is a bug in Scriban, not in NTypewriter. You could as on their [github site](https://github.com/scriban/scriban).
